src/AllowedRelationshipFilter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Spatie\QueryBuilder;
+
+use Illuminate\Support\Collection;
+use Spatie\QueryBuilder\Contracts\AllowedFilterContract;
+
+class AllowedRelationshipFilter implements AllowedFilterContract
+{
+    /** @var string */
+    protected $relationship;
+
+    /** @var \Illuminate\Support\Collection */
+    protected $allowedFilters;
+
+    public function __construct(string $relationship, AllowedFilterContract ...$allowedFilters)
+    {
+        $this->relationship = $relationship;
+
+        $this->allowedFilters = collect($allowedFilters);
+    }
+
+    public static function group(string $relationship, AllowedFilterContract ...$allowedFilters): self
+    {
+        return new static($relationship, ...$allowedFilters);
+    }
+
+    public function filter(QueryBuilder $query, $value)
+    {
+        $query->whereHas($this->relationship, function ($query) use ($value) {
+            $this->allowedFilters->each(
+                function (AllowedFilterContract $allowedFilter) use ($query, $value) {
+                    $allowedFilter->filter(
+                        QueryBuilder::for($query),
```

Comment:
I might be missing something, but why not pass the current `$query` object? It should already be an instance of `QueryBuilder`

Reply:
Tried to swap to `$query` and get error.

`Spatie\QueryBuilder\AllowedFilter::filter(): Argument #1 ($query) must be of type Spatie\QueryBuilder\QueryBuilder, Illuminate\Database\Eloquent\Builder given, called in /src/AllowedRelationshipFilter.php on line 27`

Keeping this as is.